install.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# Exit on any command failure
+set -e
+
+# Executes a command with a banner
+execute_cmd() {
+    local cmd="$@"
+    local middle_line="Command: $cmd"
+    local middle_line_len=$(echo -n "$middle_line" | wc -m)
+    local extra_chars=20
+    local total_length=$((middle_line_len + extra_chars))
+    local hash_line=$(printf '%*s' "$total_length" | tr ' ' '=')
+
+    printf "\n"
+    printf "%s\n" "$hash_line"
+    printf "          %s\n" "$middle_line"
+    printf "%s\n" "$hash_line"
+    printf "\n"
+
+    # Execute the command and check for errors
+    if ! eval "$cmd"; then
+        echo "Command failed: $cmd"
+        exit 1
+    fi
+}
+
+# Check if the script is running with root privileges
+if [ "$(id -u)" -ne 0 ]; then
+    echo "This script requires root privileges. Please run with sudo/root."
+    exit 1
+fi
+
+# Detect the operating system using uname
+OS="$(uname -s)"
+
+INSTALL=""
+
+# Perform necessary setup and get the
+# installation command depending on
+# the OS
+case "$OS" in
+    Linux*)
+        # Detect the package manager
+        PM=""
+        if command -v apt-get &> /dev/null; then
+            PM="apt-get"
+            execute_cmd apt-get -y update
+            execute_cmd apt-get -y dist-upgrade
+        elif command -v yum &> /dev/null; then
+            PM="yum"
+            execute_cmd yum -y update
+        elif command -v dnf &> /dev/null; then
+            PM="dnf"
+            execute_cmd dnf -y upgrade
+        else
+            echo "No known package manager found"
+            exit 1
+        fi
+        INSTALL="$PM install -y"
+        ;;
+    Darwin*)
+        # Check if Homebrew is installed; install it if not
+        if ! command -v brew &> /dev/null; then
+            echo "Homebrew not found. Installing Homebrew..."
```

Comment:
Maybe it's better not to do this, avoid causing as many problems as we can lol

Reply:
Commented out